packages/token-bridge-sdk/src/hooks/useBalance.spec.tsx

```diff
@@ -0,0 +1,147 @@
+  afterEach(() => {
+    jest.restoreAllMocks()
+  })
+
+  it('getter return null for undefined walletAddress', async () => {
+    const provider = new StaticJsonRpcProvider(
+      process.env.REACT_APP_ETHEREUM_RPC_URL,
+      1
+    )
+
+    const { result, waitForNextUpdate } = await renderHook(() =>
+      useBalance({
+        provider,
+        walletAddress: undefined
+      })
+    )
+
+    try {
+      await waitForNextUpdate({ timeout: 100 })
+    } catch (err) {}
+
+    const {
+      current: {
+        eth: [ethBalance]
+      }
+    } = result
+
+    expect(ethBalance).toBeNull()
+  })
+
+  it('getter return null for missing chainId', async () => {
+    const provider = new StaticJsonRpcProvider(
+      process.env.REACT_APP_ETHEREUM_RPC_URL
+    )
+
+    // @ts-ignore
+    jest.spyOn(provider, 'getNetwork').mockImplementation(() => {
+      return null
+    })
+
+    const { result, waitForNextUpdate } = await renderHook(() =>
+      useBalance({
+        provider,
+        walletAddress: '0x58b6A8A3302369DAEc383334672404Ee733aB239'
+      })
+    )
+
+    try {
+      await waitForNextUpdate({ timeout: 100 })
+    } catch (err) {}
+
+    const {
+      current: {
+        eth: [ethBalance]
+      }
+    } = result
+
+    expect(ethBalance).toBeNull()
+  })
+
+  it('getter return balance for valid tuple (walletAddress, chainId)', async () => {
+    const provider = new StaticJsonRpcProvider(
+      process.env.REACT_APP_ETHEREUM_RPC_URL,
+      1
+    )
+
+    jest.spyOn(provider, 'getBalance').mockImplementation(() => {
+      return Promise.resolve(BigNumber.from(10))
+    })
+
+    const { result, waitForNextUpdate } = await renderHook(
+      () =>
+        useBalance({
+          provider,
+          walletAddress: '0x58b6A8A3302369DAEc383334672404Ee733aB239'
+        }),
+      { wrapper: Container }
+    )
+
+    try {
+      await waitForNextUpdate({ timeout: 100 })
+    } catch (err) {}
+
+    const {
+      current: {
+        eth: [ethBalance]
+      }
+    } = result
+
+    expect(ethBalance?.toNumber()).toEqual(10)
+  })
+
+  it('setter update balance ', async () => {
+    const provider = new StaticJsonRpcProvider(
+      process.env.REACT_APP_ETHEREUM_RPC_URL,
+      1
+    )
+    jest.spyOn(provider, 'getBalance').mockImplementation(() => {
+      return Promise.resolve(BigNumber.from(30))
+    })
+
+    const { result, waitForNextUpdate } = await renderHook(() =>
+      useBalance({
+        provider,
+        walletAddress: '0x58b6A8A3302369DAEc383334672404Ee733aB239'
+      })
+    )
+
+    try {
+      await waitForNextUpdate({ timeout: 100 })
+    } catch (err) {}
+
+    const {
+      current: {
+        eth: [ethBalance, updateEthBalance]
+      }
+    } = result
+
+    expect(ethBalance?.toNumber()).toEqual(30)
+
+    await act(async () => {
+      await updateEthBalance()
+
+      try {
+        await waitForNextUpdate({ timeout: 500 })
+        expect(ethBalance?.toNumber()).toEqual(40)
```

Comment:
Hm, I must be missing something, but how did we get from `30` to `40` here? 😄

Reply:
Good catch 😄, I have updated the test